plugins/inputs/suricata/README.md

```diff
@@ -0,0 +1,135 @@
+engine, such as captured traffic volume, memory usage, uptime, flow counters,
+and much more. It provides a socket for the Suricata log output to write JSON
+stats output to, and processes the incoming data to fit Telegraf's format.
+
+### Configuration:
+
+```toml
+[[input.suricata]]
+  ## Data sink for Suricata stats log.
+  # This is expected to be a filename of a
+  # unix socket to be created for listening.
+  # Will be overwritten if a socket or file
+  # with that name already exists.
+  source = "/tmp/suricata-stats.sock"
+```
+
+### Measurements & Fields:
+
+Fields in the 'suricata' measurement follow the JSON format used by Suricata's
+stats output.
+See http://suricata.readthedocs.io/en/latest/performance/statistics.html for
+more information.
+
+All fields are numeric.
+
+- suricata
+    - app_layer.flow.dcerpc_udp
+    - app_layer.flow.dns_tcp
+    - app_layer.flow.dns_udp
+    - app_layer.flow.enip_udp
+    - app_layer.flow.failed_tcp
+    - app_layer.flow.failed_udp
+    - app_layer.flow.http
+    - app_layer.flow.ssh
+    - app_layer.flow.tls
+    - app_layer.tx.dns_tcp
+    - app_layer.tx.dns_udp
+    - app_layer.tx.enip_udp
+    - app_layer.tx.http
+    - app_layer.tx.smtp
+    - capture.kernel_drops
+    - capture.kernel_packets
+    - decoder.avg_pkt_size
+    - decoder.bytes
+    - decoder.ethernet
+    - decoder.gre
+    - decoder.icmpv4
+    - decoder.icmpv4.ipv4_unknown_ver
+    - decoder.icmpv6
+    - decoder.invalid
+    - decoder.ipv4
+    - decoder.ipv6
+    - decoder.max_pkt_size
+    - decoder.pkts
+    - decoder.tcp
+    - decoder.tcp.hlen_too_small
+    - decoder.tcp.invalid_optlen
+    - decoder.teredo
+    - decoder.udp
+    - decoder.vlan
+    - detect.alert
+    - dns.memcap_global
+    - dns.memuse
+    - flow.memuse
+    - flow_mgr.closed_pruned
+    - flow_mgr.est_pruned
+    - flow_mgr.flows_checked
+    - flow_mgr.flows_notimeout
+    - flow_mgr.flows_removed
+    - flow_mgr.flows_timeout
+    - flow_mgr.flows_timeout_inuse
+    - flow_mgr.new_pruned
+    - flow_mgr.rows_checked
+    - flow_mgr.rows_empty
+    - flow_mgr.rows_maxlen
+    - flow_mgr.rows_skipped
+    - flow.spare
+    - flow.tcp_reuse
+    - http.memuse
+    - tcp.memuse
+    - tcp.pseudo
+    - tcp.reassembly_gap
+    - tcp.reassembly_memuse
+    - tcp.rst
+    - tcp.sessions
+    - tcp.syn
+    - tcp.synack
+    - ...
+
+
+### Tags:
+
+The `suricata` measurement has the following tags:
+
+- thread: `Global` for global statistics, thread IDs (e.g. `W#03-enp0s31f6`) for thread-specific statistics, `total` for a summary across all threads.
+
+## Suricata configuration
+
+Suricata needs to deliver the 'stats' event type to a given unix socket for
+this plugin to pick up. This can be done, for example, by creating an additional
+output in the Suricata configuration file:
+
+```yaml
+- eve-log:
+    enabled: yes
+    filetype: unix_stream
+    filename: /tmp/suricata-stats.sock
+    types:
+      - stats:
+         totals: yes
```

Comment:
I am not completely sure what Suricata does there , but I'd suspect that's what these are, yes.
I can just remove that line from the README if it complicates things.

Reply:
Yeah do that, we generally try to avoid collecting aggregate data, unless we consider it an "alternative view" of the data, such as if it is provided by a different source of truth.  If it's just adding up the threads for us it will be more flexible to turn it off, and the aggregation can be done at query time.